File: jetbrains/src/main/kotlin/deplens/utils/resolver/DartImportResolver.kt
package deplens.utils.resolver

object DartImportResolver {

    private val importPattern = Regex("""^\s*import\s+['"]([^'"]+)['"]""")

    fun getDepName(line: String): String? {
        return importPattern.find(line)?.groupValues?.getOrNull(1)
    }

    fun isLocalImport(dep: String): Boolean {
        return dep.startsWith(".") || dep.startsWith("/") || dep.startsWith("dart:")
    }

    fun isPackageImport(dep: String): Boolean {
        return dep.startsWith("package:")
    }

    fun getPkgName(dep: String): String {
        val normalized = dep.removePrefix("package:")
        return normalized.substringBefore("/")
    }
}

File: vscode/src/utils/resolver/DartImportResolver.ts
export class DartImportResolver {
  static isImport(line: string): boolean {
    return /^\s*import\s+['"]([^'"]+)['"]/.test(line);
  }

  static getDepName(line: string): string | null {
    const match = line.match(/^\s*import\s+['"]([^'"]+)['"]/);
    return match?.[1] ?? null;
  }

  static isLocalImport(dep: string): boolean {
    return dep.startsWith(".") || dep.startsWith("/") || dep.startsWith("dart:");
  }

  static isPackageImport(dep: string): boolean {
    return dep.startsWith("package:");
  }

  static getPkgName(dep: string): string {
    const normalized = dep.startsWith("package:") ? dep.slice("package:".length) : dep;
    return normalized.split("/")[0];
  }
}
